torch/csrc/profiler/python/init.cpp

```diff
@@ -210,8 +213,17 @@ PyObject* RecordFunctionFast_enter(PyObject* selfGeneric, PyObject* unused) {
         std::make_unique<at::RecordFunction>(at::RecordScope::FUNCTION);
     if (self->inputValues) {
       std::vector<at::IValue> ivalues;
-      for (int i = 0; i < PyList_Size(self->inputValues); i++) {
-        PyObject* item = PyList_GetItem(self->inputValues, i);
+      std::function<Py_ssize_t(PyObject*)> getSize;
+      std::function<PyObject*(PyObject*, Py_ssize_t)> getItem;
+      if (PyTuple_Check(self->inputValues)) {
+        getSize = PyTuple_Size;
+        getItem = PyTuple_GetItem;
+      } else {
+        getSize = PyList_Size;
+        getItem = PyList_GetItem;
+      }
+      for (int i = 0; i < getSize(self->inputValues); i++) {
+        PyObject* item = getItem(self->inputValues, i);
```

Comment:
How about:
```suggestion
      THPObjectPtr input_fast(PySequence_Fast(self->inputValues, "input must be a sequence"));
      PyObject ** input_items = PySequence_Fast_ITEMS(input_fast);
      for (int i = 0; i < PySequence_Fast_GET_SIZE(input_fast); i++) {
        PyObject* item = input_items[i];
```

Reply:
Thanks for the suggestion! Let me try this out.